threads/main.go

```diff
@@ -0,0 +1,196 @@
+
+	blocks := strings.Split(in, " ")
+	switch blocks[0] {
+	case "use":
+		if len(blocks) < 2 {
+			fmt.Println("You must provide a store ID.")
+			return
+		}
+		currentStore = blocks[1]
+		fmt.Printf("Switched to %s\n", blocks[1])
+		return
+	case "listen":
+		listen(currentStore)
+		return
+	case "getModel":
+		if len(blocks) < 1 {
+			fmt.Println("You must provide a model name.")
+			return
+		}
+		getModel(currentStore, blocks[1], apiTimeout)
+		return
+	case "exit":
+		fmt.Println("Bye!")
+		os.Exit(0)
+	}
+	fmt.Println("Sorry, I don't understand.")
+}
+
+func getModel(id string, model string, maxAwaitTime time.Duration) {
+	ctx, cancel := context.WithTimeout(context.Background(), apiTimeout)
+	defer cancel()
+
+	rawResults, err := apiClient.ModelFind(ctx, id, model, nil, []*any{})
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	entities := rawResults.([]*any)
+	if len(entities) == 0 {
+		fmt.Printf("None found")
+		return
+	}
+	for _, el := range entities {
+		prettyPrint(el)
+	}
+}
+
+func listen(id string) {
+	channel, err := apiClient.Listen(context.Background(), id)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	for {
+		select {
+		case val, ok := <-channel:
+			if !ok {
+				return
+			} else {
+				if val.Err != nil {
+					fmt.Println(val.Err)
+					continue
+				}
+				obj := &any{}
+				if err := json.Unmarshal(val.Action.Entity, obj); err != nil {
+					fmt.Println("failed to unmarshal listen result")
+					continue
+				}
+				prettyPrint(obj)
+			}
+		case <-time.After(streamTimeout):
+			fmt.Println("timeout")
+			return
+		}
+	}
+}
+
+func main() {
+	addr := flag.String("addr", "/ip4/0.0.0.0/tcp/6006", "Threads APIs")
+	flag.Parse()
+	maddr, err := ma.NewMultiaddr(*addr)
+	if err != nil {
+		log.Fatalf("addr error: %v", err)
+	}
+
+	target, err := util.TCPAddrFromMultiAddr(maddr)
+	if err != nil {
+		log.Fatalf("addr error: %v", err)
+	}
+
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithInsecure())
+	opts = append(opts, grpc.WithBlock())
+	conn, err := grpc.Dial(target, opts...)
+	if err != nil {
+		log.Fatalf("connection error: %v", err)
+	}
+	defer conn.Close()
+
+	apiClient, err = client.NewClient(target, opts...)
+	if err != nil {
+		log.Fatalf("connection error: %v", err)
+	}
+
+	fmt.Println("Successfully connected.")
+
+	p := prompt.New(
+		executor,
+		completer,
+		prompt.OptionPrefix(">>> "),
+
+		prompt.OptionPreviewSuggestionTextColor(prompt.Green),
+		prompt.OptionPreviewSuggestionBGColor(prompt.Black),
+
+		prompt.OptionSelectedSuggestionTextColor(prompt.White),
+		prompt.OptionSelectedSuggestionBGColor(prompt.DarkGray),
+
+		prompt.OptionSuggestionTextColor(prompt.LightGray),
+		prompt.OptionSuggestionBGColor(prompt.Black),
+
+		prompt.OptionDescriptionTextColor(prompt.Turquoise),
+		prompt.OptionDescriptionBGColor(prompt.Black),
+		prompt.OptionSelectedDescriptionTextColor(prompt.Turquoise),
+		prompt.OptionSelectedDescriptionBGColor(prompt.Black),
```

Comment:
the defaults were a bit hard to look at after a while, so just toned them down here. not tied to these though